File: codes/python/chap05/answer_A32.py
# 입력
N, A, B = map(int, input().split())

# 배열 dp 정의
# dp[x]=True일 때 승리 상태, dp[x]=False일 때 패배 상태
dp = [ None ] * (N+1)

# 승자를 계산한다
for i in range(N+1):
	if i >= A and dp[i-A] == False:
		dp[i] = True
	elif i >= B and dp[i-B] == False:
		dp[i] = True
	else:
		dp[i] = False

# 출력
if dp[N] == True:
	print("First")
else:
	print("Second")

File: codes/cpp/chap05/answer_A32.cpp
#include <iostream>
using namespace std;

// 배열 dp에 대해: dp[x]=true일 때 승리 상태, dp[x]=false일 때 패배 상태
int N, A, B;
bool dp[100009];

int main() {
	// 입력
	cin >> N >> A >> B;

	// 승자를 계산한다
	for (int i = 0; i <= N; i++) {
		if (i >= A && dp[i - A] == false) dp[i] = true; // 승리 상태
		else if (i >= B && dp[i - B] == false) dp[i] = true; // 승리 상태
		else dp[i] = false; // 패배 상태
	}

	// 출력
	if (dp[N] == true) cout << "First" << endl;
	else cout << "Second" << endl;
	return 0;
}
